src/Interactive/EditorFeatures/Core/Extensibility/Interactive/InteractiveEvaluator.cs

```diff
@@ -86,7 +85,7 @@ internal abstract class InteractiveEvaluator : IInteractiveEvaluator, ICurrentWo
 
             _contentType = contentType;
             _responseFilePath = responseFilePath;
-            _workspace = new InteractiveWorkspace(hostServices);
+            _workspace = new InteractiveWorkspace(new CustomizedHostServices(hostServices, GetInteractiveWorkspaceService));
```

Comment:
Is the whole point of this to avoid exporting a class that implements `IScriptEnvironmentService` with `WorkspaceKind.Interactive` (like [so](http://source.roslyn.io/#Microsoft.CodeAnalysis.InteractiveEditorFeatures/Implementation/Interactive/InteractiveDocumentSupportsSuggestionService.cs,12)), since it can't be initialized with the correct values?

Reply:
I didn't realized this can be done for both misc and interactive workspaces. Will try that.

---
In reply to: [136115110](https://github.com/dotnet/roslyn/pull/21487#discussion_r136115110) [](ancestors = 136115110)